packages/editor/src/components/rich-text/format-toolbar/link-container.js

```diff
@@ -142,7 +147,13 @@ class LinkContainer extends Component {
 		event.preventDefault();
 	}
 
-	resetState() {
+	resetState( event ) {
+		// URLInput's autocomplete list can trigger onClickOutside on the link container's popover.
+		// Return early here if the click originated from one of the autocomplete suggestions.
+		if ( hasIn( event, [ 'target', 'classList' ] ) && includes( event.target.classList, 'editor-url-input__suggestion' ) ) {
```

Comment:
It'd be nice to be able to avoid checking the class here, but I'm short of other ideas beyond having LinkContainer manage a ref to the URLInput's popover.

Reply:
Yes this is a bit unfortunate.